src/overlays/actors/ovl_Boss_04/z_boss_04.c

```diff
@@ -69,75 +74,776 @@ const ActorInit Boss_04_InitVars = {
+                this->unk_2D0 = 0.0f;
+                D_809EE4D0 = 1;
+                this->unk_2E2 = 60;
+                this->unk_2E0 = 93;
+            }
+
+            if (this->unk_704 > 140) {
+                Camera* sp5C = Play_GetCamera(globalCtx, MAIN_CAM);
+                this->unk_708 = 0;
+                func_809ECD00(this, globalCtx);
+                sp5C->eye = this->unk_70C;
+                sp5C->eyeNext = this->unk_70C;
+                sp5C->at = this->unk_718;
+                func_80169AFC(globalCtx, this->unk_70A, 0);
+                this->unk_70A = 0;
+                func_800EA0EC(globalCtx, &globalCtx->csCtx);
+                func_800B7298(globalCtx, &this->actor, 6);
+                func_80165690();
+                gSaveContext.eventInf[6] |= 1;
+            }
+            break;
+    }
+
+    if (this->unk_70A != 0) {
+        Vec3f sp50;
+
+        ShrinkWindow_SetLetterboxTarget(0x1B);
+        if (this->unk_748 != 0) {
+            this->unk_748--;
+        }
+        Math_Vec3f_Copy(&sp50, &this->unk_718);
+        sp50.y += Math_SinS(this->unk_748 * 0x4000) * this->unk_748 * 1.5f;
+        Play_CameraSetAtEye(globalCtx, this->unk_70A, &sp50, &this->unk_70C);
+        func_80169940(globalCtx, this->unk_70A, this->unk_744);
+        Math_ApproachF(&this->unk_744, 60.0f, 0.1f, 1.0f);
+    }
+    this->actor.shape.rot.y = this->actor.yawTowardsPlayer;
+    x = player->actor.world.pos.x - this->actor.world.pos.x;
+    y = player->actor.world.pos.y - this->actor.world.pos.y;
+    z = player->actor.world.pos.z - this->actor.world.pos.z;
+    this->actor.shape.rot.x = Math_Atan2S(-(y), sqrtf(SQ(x) + SQ(z)));
+}
+
+void func_809ECD00(Boss04* this, GlobalContext* globalCtx) {
+    this->actionFunc = func_809ECD18;
+}
+
+void func_809ECD18(Boss04* this, GlobalContext* globalCtx) {
+    Player* player = GET_PLAYER(globalCtx);
+
+    if (this->unk_1F4 >= 0x500) {
+        this->unk_1F4 -= 0x300;
+        this->unk_1F7 += 1;
+    }
+
+    Math_ApproachS(&this->actor.shape.rot.y, this->actor.yawTowardsPlayer, 10, 0x200);
+    this->actor.world.pos.y = (this->actor.floorHeight + KREG(17) + 160.0f) + (Math_SinS(this->unk_1F4 * 512) * 10.0f);
+    Math_ApproachF(&this->actor.speedXZ, this->unk_6D4, 1.0f, 0.5f);
+
+    if (this->unk_1F8 == 0) {
+        this->unk_1F8 = Rand_ZeroFloat(100.0f) + 50.0f;
+        this->unk_6D4 = Rand_ZeroFloat(3.0f);
+        if (Rand_ZeroOne() < 0.1f) {
+            Math_Vec3f_Copy(&this->unk_6C8, &player->actor.world.pos);
+        } else {
+            this->unk_6C8.x = randPlusMinusPoint5Scaled(600.0f) + this->unk_6E8;
+            this->unk_6C8.z = randPlusMinusPoint5Scaled(600.0f) + this->unk_6F0;
+        }
+    }
+
+    Math_ApproachS(&this->actor.world.rot.y,
+                   Math_Atan2S(this->unk_6C8.x - this->actor.world.pos.x, this->unk_6C8.z - this->actor.world.pos.z), 5,
+                   0x200);
+
+    if (((s8)this->actor.colChkInfo.health < 11) || (KREG(88) != 0)) {
```

Comment:
```suggestion
    if (((s8)this->actor.colChkInfo.health <= 10) || (KREG(88) != 0)) {
```
Small nit, but round number is probably preferred.